<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>KeyboardLayoutEditor</class>
 <widget class="QWidget" name="KeyboardLayoutEditor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>614</width>
    <height>480</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="m_verticalLayout">
   <property name="spacing">
    <number>2</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="KMessageWidget" name="m_messageWidget"/>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QToolButton" name="m_newKeyToolButton">
       <property name="toolTip">
        <string>Create new normal key</string>
       </property>
       <property name="text">
        <string>Normal Key</string>
       </property>
       <property name="icon">
        <iconset theme="list-add">
         <normaloff/>
        </iconset>
       </property>
       <property name="toolButtonStyle">
        <enum>Qt::ToolButtonTextBesideIcon</enum>
       </property>
       <property name="autoRaise">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="m_newSpecialKeyToolButton">
       <property name="toolTip">
        <string>Create new special key</string>
       </property>
       <property name="text">
        <string>Special Key</string>
       </property>
       <property name="icon">
        <iconset theme="list-add">
         <normaloff/>
        </iconset>
       </property>
       <property name="toolButtonStyle">
        <enum>Qt::ToolButtonTextBesideIcon</enum>
       </property>
       <property name="autoRaise">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="m_deleteKeyToolButton">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="toolTip">
        <string>Delete selected key</string>
       </property>
       <property name="text">
        <string>Delete</string>
       </property>
       <property name="icon">
        <iconset theme="edit-delete">
         <normaloff/>
        </iconset>
       </property>
       <property name="toolButtonStyle">
        <enum>Qt::ToolButtonTextBesideIcon</enum>
       </property>
       <property name="autoRaise">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="Line" name="line">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="m_zoomOutToolButton">
       <property name="icon">
        <iconset theme="zoom-out">
         <normaloff/>
        </iconset>
       </property>
       <property name="autoRaise">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSlider" name="m_zoomSlider">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>100</width>
         <height>0</height>
        </size>
       </property>
       <property name="minimum">
        <number>-4</number>
       </property>
       <property name="maximum">
        <number>4</number>
       </property>
       <property name="pageStep">
        <number>1</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="m_zoomFactorLabel">
       <property name="text">
        <string>100%</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QToolButton" name="m_zoomInToolButton">
       <property name="icon">
        <iconset theme="zoom-in">
         <normaloff/>
        </iconset>
       </property>
       <property name="autoRaise">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="m_horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item>
    <widget class="KeyboardLayoutEditorView" name="m_view">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="verticalScrollBarPolicy">
      <enum>Qt::ScrollBarAsNeeded</enum>
     </property>
     <property name="horizontalScrollBarPolicy">
      <enum>Qt::ScrollBarAsNeeded</enum>
     </property>
     <property name="backgroundBrush">
      <brush brushstyle="SolidPattern">
       <color alpha="255">
        <red>88</red>
        <green>88</green>
        <blue>88</blue>
       </color>
      </brush>
     </property>
    </widget>
   </item>
   <item>
    <widget class="KeyboardLayoutPropertiesWidget" name="m_propertiesWidget" native="true">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>KMessageWidget</class>
   <extends>QFrame</extends>
   <header location="global">kmessagewidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>KeyboardLayoutEditorView</class>
   <extends>QGraphicsView</extends>
   <header>editor/keyboardlayouteditorview.h</header>
  </customwidget>
  <customwidget>
   <class>KeyboardLayoutPropertiesWidget</class>
   <extends>QWidget</extends>
   <header>editor/keyboardlayoutpropertieswidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>m_view</tabstop>
  <tabstop>m_newKeyToolButton</tabstop>
  <tabstop>m_newSpecialKeyToolButton</tabstop>
  <tabstop>m_deleteKeyToolButton</tabstop>
  <tabstop>m_zoomOutToolButton</tabstop>
  <tabstop>m_zoomSlider</tabstop>
  <tabstop>m_zoomInToolButton</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>m_zoomOutToolButton</sender>
   <signal>clicked()</signal>
   <receiver>m_zoomSlider</receiver>
   <slot>subtractStep()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>307</x>
     <y>28</y>
    </hint>
    <hint type="destinationlabel">
     <x>354</x>
     <y>26</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_zoomInToolButton</sender>
   <signal>clicked()</signal>
   <receiver>m_zoomSlider</receiver>
   <slot>addStep()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>470</x>
     <y>30</y>
    </hint>
    <hint type="destinationlabel">
     <x>411</x>
     <y>27</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
